sequencer/common/zk.go

```diff
@@ -172,36 +107,29 @@ type ZKInputs struct {
 	// leaf).
 	// If Tx is an Exit (tx.ToIdx=1), state 2 is used for the Exit Merkle
 	// Proof of the Exit MerkleTree.
-	TokenID2  []*big.Int   `json:"tokenID2"`  // uint32, len: [maxTx]
+	// TokenID2  []*big.Int   `json:"tokenID2"`  // uint32, len: [maxTx]
 	Nonce2    []*big.Int   `json:"nonce2"`    // uint64 (max 40 bits), len: [maxTx]
-	Sign2     []*big.Int   `json:"sign2"`     // bool, len: [maxTx]
+	Sign2     []*bool      `json:"sign2"`     // bool, len: [maxTx]
 	Ay2       []*big.Int   `json:"ay2"`       // big.Int, len: [maxTx]
 	Balance2  []*big.Int   `json:"balance2"`  // big.Int (max 192 bits), len: [maxTx]
 	EthAddr2  []*big.Int   `json:"ethAddr2"`  // ethCommon.Address, len: [maxTx]
 	Siblings2 [][]*big.Int `json:"siblings2"` // big.Int, len: [maxTx][nLevels + 1]
-	// NewExit determines if an exit transaction has to create a new leaf
-	// in the exit tree. If already exists an exit leaf of an account in
-	// the ExitTree, there is no 'new leaf' creation and 'NewExit' for that
-	// tx is 0 (if is an 'insert' in the tree, NewExit=1, if is an 'update'
-	// of an existing leaf, NewExit=0).
-	NewExit []*big.Int `json:"newExit"` // bool, len: [maxTx]
 	// Required for inserts and deletes, values of the CircomProcessorProof
 	// (smt insert proof)
-	IsOld0_2  []*big.Int `json:"isOld0_2"`  // bool, len: [maxTx]
+	IsOld0_2  []*bool    `json:"isOld0_2"`  // bool, len: [maxTx]
 	OldKey2   []*big.Int `json:"oldKey2"`   // uint64 (max 40 bits), len: [maxTx]
 	OldValue2 []*big.Int `json:"oldValue2"` // Hash, len: [maxTx]
 
-	// state 3, fee leafs states, value of the account leaf receiver of the
-	// Fees fee tx. The values at the moment pre-smtprocessor of the update
-	// (before updating the Receiver leaf).
-	// The order of FeeIdxs & FeePlanTokens & State3 must match.
-	TokenID3  []*big.Int   `json:"tokenID3"`  // uint32, len: [maxFeeIdxs]
-	Nonce3    []*big.Int   `json:"nonce3"`    // uint64 (max 40 bits), len: [maxFeeIdxs]
-	Sign3     []*big.Int   `json:"sign3"`     // bool, len: [maxFeeIdxs]
-	Ay3       []*big.Int   `json:"ay3"`       // big.Int, len: [maxFeeIdxs]
-	Balance3  []*big.Int   `json:"balance3"`  // big.Int (max 192 bits), len: [maxFeeIdxs]
-	EthAddr3  []*big.Int   `json:"ethAddr3"`  // ethCommon.Address, len: [maxFeeIdxs]
-	Siblings3 [][]*big.Int `json:"siblings3"` // Hash, len: [maxFeeIdxs][nLevels + 1]
+	//Vouch
+	VouchExists   []*bool      `json:"vouchExists"`   // bool, len: [maxTx]
+	VouchSiblings [][]*big.Int `json:"vouchSiblings"` // len: [maxTx][2*nLevel + 1]
+
+	//Score, 1. Sender, 2. Reciever
+	Score1         []*uint32    `json:"score1"`         // uint32, len: [maxTx]
```

Comment:
how about renaming this to `SenderScore`

Reply:
We could rename it, but the same convention is followed by circuit team on there end hence will need to sync it with them